File: src/test/java/com/pschlup/ta/timeseries/BarTest.java
package com.pschlup.ta.timeseries;

import org.junit.jupiter.api.Test;

import java.time.Instant;

import static com.google.common.truth.Truth.assertThat;

class BarTest {

  @Test
  void shouldIncludeTickTimestamp() {
    Instant openTime = Instant.parse("2019-04-19T20:00:00Z");
    Bar bar = new Bar(TimeFrame.M5, openTime, 0.);
    Instant tickTime = Instant.parse("2019-04-19T20:04:15Z");
    assertThat(bar.includesTimestamp(tickTime)).isTrue();
  }

  @Test
  void shouldNotIncludeNextOpenTimestamp() {
    Instant openTime = Instant.parse("2019-04-19T20:00:00Z");
    Bar bar = new Bar(TimeFrame.M5, openTime, 0.);
    Instant nextOpen = Instant.parse("2019-04-19T20:05:00Z");
    assertThat(bar.includesTimestamp(nextOpen)).isFalse();
  }
}

File: src/test/kotlin/com/pschlup/ta/timeseries/BarTest.kt
package com.pschlup.ta.timeseries

import com.google.common.truth.Truth.assertThat
import org.junit.jupiter.api.Test
import java.time.Instant

internal class BarTest {
  @Test
  fun shouldIncludeTickTimestamp() {
    val openTime = Instant.parse("2019-04-19T20:00:00Z")
    val bar = Bar(TimeFrame.M5, openTime, 0.0)
    val tickTime = Instant.parse("2019-04-19T20:04:15Z")

    assertThat(bar.includesTimestamp(tickTime)).isTrue()
  }

  @Test
  fun shouldNotIncludeNextOpenTimestamp() {
    val openTime = Instant.parse("2019-04-19T20:00:00Z")
    val bar = Bar(TimeFrame.M5, openTime, 0.0)
    val nextOpen = Instant.parse("2019-04-19T20:05:00Z")

    assertThat(bar.includesTimestamp(nextOpen)).isFalse()
  }
}
